### PR TITLE
first implementation of terraform-aws-public-route53-zone

### PR DESCRIPTION
**Pull request by salvianreynaldi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,21 @@
...
+
+## Conventions
+
+## Authors
+  - [Salvian Reynaldi](https://github.com/salvianreynaldi)
+
+
+## License
+
+GNU General Public License v3.0. See LICENSE for full details.
```
Comment (andysaputra) [inline README.md]: Should be Apache

File: README.md [hunk 2]
```diff
@@ -1,2 +1,21 @@
...
+## Usage
+```hcl
+module "public-route53-hosted-zone" {
+    source = "github.com/traveloka/terraform-public-route53-zone?ref=master"
+    name = "flight.traveloka.com"
+}
+
+```
+
+## Conventions
```
Comment (andysaputra) [inline README.md]: What is this section for?

Comment (salvianreynaldi) [reply README.md]: this section is to list any restrictions / conventions that this module impose, e.g. naming, tagging

File: README.md [hunk 3]
```diff
@@ -1,2 +1,21 @@
-# terraform-aws-public-route53-zone
-A terraform module which provisions a Public Route53 Hosted Zone
+# terraform-public-route53-zone
```
Comment (sshvetsov) [inline README.md]: I think 'aws' should stay in the name a per module naming guidelines

File: README.md [hunk 4]
```diff
@@ -1,2 +1,21 @@
...
+A Terraform module which creates a public AWS route53 hosted zone
+
+## Usage
+```hcl
+module "public-route53-hosted-zone" {
+    source = "github.com/traveloka/terraform-public-route53-zone?ref=master"
```
Comment (sshvetsov) [inline README.md]: avoid using "master" in examples, always use explicit module version, or iser will have an unpleasant surprise during backwards incompatible module upgrade.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,21 @@
...
+    name = "flight.traveloka.com"
```
Comment (sshvetsov) [inline README.md]: let's use more realistic example, like "fpr.traveloka.com"

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,7 @@
+resource "aws_route53_zone" "main" {
```
Comment (sshvetsov) [inline main.tf]: rename to 'this'

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,24 @@
+variable "name" {
```
Comment (sshvetsov) [inline variables.tf]: Apply sentence case to all descriptions.

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  default     = ""
+  description = "the delegation set ID whose NS records will be assigned the hosted zone"
+}
+
+variable "force_destroy" {
+  default     = false
+  description = "whether to destroy all records inside if the hosted zone is deleted"
+}
+
+variable "tags" {
```
Comment (sshvetsov) [inline variables.tf]: Let's remove tags variable and instead request required tag values as per conventions.

File: main.tf [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+  name              = "${var.name}"
+  comment           = "${var.comment}"
+  delegation_set_id = "${var.delegation_set_id}"
+  force_destroy     = "${var.force_destroy}"
+  tags              = "${var.tags}"
```
Comment (sshvetsov) [inline main.tf]: Best to define tags in the module from the provided variables to enforce conventions.

File: outputs.tf [hunk 10]
```diff
@@ -0,0 +1,4 @@
+output "zone_id" {
+  value       = "${aws_route53_zone.main.zone_id}"
+  description = "the hosted zone id"
```
Comment (sshvetsov) [inline outputs.tf]: Apply sentence case to descriptions.

File: examples/flight/main.tf [hunk 11]
```diff
@@ -0,0 +1,12 @@
+provider "aws" {
+  region = "ap-southeast-1"
+}
+
+module "public-route53-hosted-zone" {
+  source = "../.."
+  name   = "flight.traveloka.com"
+
+  tags = {
```
Comment (sshvetsov) [inline examples/flight/main.tf]: Provide a better example, one reflecting conventions.

File: variables.tf [hunk 12]
```diff
@@ -0,0 +1,32 @@
+variable "tag_name" {
+  type        = "string"
+  description = "Name of the hosted zone"
+}
+
```
Comment (andysaputra) [inline variables.tf]: I propose to remove the `tag_` prefix for the parameters below

File: README.md [hunk 13]
```diff
@@ -1,2 +1,25 @@
 # terraform-aws-public-route53-zone
-A terraform module which provisions a Public Route53 Hosted Zone
+A Terraform module which creates a public AWS route53 hosted zone
+
+## Usage
+```hcl
+module "public-route53-hosted-zone" {
+    source = "github.com/traveloka/terraform-public-route53-zone?ref=0.1.0"
+    name = "fpr.traveloka.com"
```
Comment (andysaputra) [inline README.md]: seems to be an incomplete example, missing some required parameters.

**Pull request merged**